Source/WebCore/bindings/js/JSTrustedTypePolicyFactoryCustom.cpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2024 Igalia S.L. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "JSTrustedTypePolicyFactory.h"
+
+#include <JavaScriptCore/SlotVisitorInlines.h>
+
+namespace WebCore {
+
+template<typename Visitor>
+void JSTrustedTypePolicyFactory::visitAdditionalChildren(Visitor& visitor)
+{
+    Locker locker { wrapped().lock() };
+    if (auto defaultPolicy = wrapped().defaultPolicy())
```

Comment:
`defaultPolicy()` returns `RefPtr<TrustedTypePolicy>`, so this would calls `ref()`, which causes a crash of thread checking. We should make `defaultPolicy()` returns `TrustedTypePolicy*`, not `RefPtr<TrustedTypePolicy>`.

Reply:
Thanks, have done this now, and it indeed fixes the issue I had with the ASAN build.